go.go

```diff
@@ -99,12 +99,26 @@ func Go(a, b []byte) (changes []*Change, err error) {
 			ds := x.diffstat(ai, bi)
 			changes = append(changes, &Change{Name: s, DelLines: ds.del, InsLines: ds.ins})
 		} else {
-			changes = append(changes, &Change{Name: s, Inserted: true, InsLines: len(bytes.Split(x.bBytes(bi), newline))})
+			var ln int
+			if len(ins) == len(del) {
+				ds := x.diffstat(bi, bi)
+				ln = ds.ins
+			} else {
+				ln = len(bytes.Split(x.bBytes(bi), newline))
```

Comment:
Now that this construction is being used even more, I wonder whether it should be a method on `bySplits`. Something like:

func (x *bySplits) bLineCount(bi int) int {
  return len(bytes.Split(x.bBytes(bi), newline))
}
```

and similar for `aLineCount(ai)`. Or could call it `aLines` or `aNumLines`.